consensus/src/chained_bft/liveness/proposal_generator.rs

```diff
@@ -83,6 +83,24 @@ impl<T: Payload> ProposalGenerator<T> {
         }
     }
 
+    /// Creates a NIL block proposal extending the highest certified block from the block store.
+    pub fn generate_nil_block(&self, round: Round) -> Result<Block<T>, ProposalGenerationError> {
+        let hqc_block = self.block_store.highest_certified_block();
+        if hqc_block.round() >= round {
```

Comment:
Why not just get block_store.highest_quorum_cert?

Reply:
Hmm we need both the block and its QC (to create a child)

Reply:
You are right!  However, this is a great opportunity to remove the parent_block from make_block().  Will file another issue for this.  https://github.com/libra/libra/issues/514